<launch>	
	<node pkg="local_planning" name="pub_goal" type="pub_goal" output="screen" launch-prefix="gnome-terminal --">		
	</node>	
	<arg name="topic_of_local_pcl" default="/scan"/>
	<!-- 启动局部规划算法 -->
	<node pkg="local_planning" name="local_planner_main" type="local_planner_main" output="screen" >	
		<remap from="/prometheus/planning/local_pcl" to="$(arg topic_of_local_pcl)" />
		<!-- 参数 -->
		<!-- 选择算法 0代表APF,1代表VFH -->
		<param name="local_planner/algorithm_mode" value="0" type="int"/>
		<!-- 激光雷达模型,0代表3d雷达,1代表2d雷达 -->
		<param name="local_planner/lidar_model" value="1" type="int"/>
		<!-- 飞行高度 -->
		<param name="local_planner/fly_height_2D" value="1.0" type="double"/>
		<param name="local_planner/sim_mode" value="true" type="bool"/>
		<!-- 最大速度，设的越小越安全 -->
		<param name="local_planner/max_planning_vel" value="0.2" type="double"/>
		<!-- 膨胀参数，一般设置为无人机的半径或更大 -->
		<param name="apf/inflate_distance" value="0.4" type="double"/>
		<!-- 感知距离，只考虑感知距离内的障碍物 -->
		<param name="apf/obs_distance" value="5.0" type="double"/>  
		<!-- 增益 -->
		<param name="apf/k_push" value="2.0" type="double"/>
		<param name="apf/k_att" value="1.0" type="double"/>
		<!-- 安全距离，此参数有问题，小于改距离会吸到障碍物上去 -->
		<param name="apf/min_dist" value="0" type="double"/>
		<!-- 最大吸引距离 -->
		<param name="apf/max_att_dist" value="4" type="double"/>
		<!-- 地面高度，不考虑低于地面高度的障碍物 -->
		<param name="apf/ground_height" value="0.3" type="double"/>
		<!-- 地面安全高度，小于该高度，会产生向上推力 -->
		<param name="apf/ground_safe_height" value="0.3" type="double"/>
		<!-- 停止距离，小于该距离，停止自动飞行 -->
		<param name="apf/safe_distance" value="0.01" type="double"/>
	</node>	
</launch>
